testing/python/collect.py

```diff
@@ -841,6 +841,33 @@ def test_makeitem_non_underscore(self, testdir, monkeypatch):
         l = modcol.collect()
         assert '_hello' not in l
 
+    def test_issue2369_collect_module_fileext(self, testdir):
+        """Ensure we can collect files with weird file extensions as Python
+        modules (#2369)"""
+        # We'll implement a little finder and loader to import files containing
+        # Python source code whose file extension is ".narf".
+        testdir.makeconftest("""
+            import sys, os, imp
+            from _pytest.python import Module
+
+            class Loader:
```

Comment:
for a different protection scheme you can use pytest_configure and pytest_unconfigure to push/pop the extra finder